src/pretix/plugins/banktransfer/payment.py

```diff
@@ -144,6 +157,10 @@ def settings_form_clean(self, cleaned_data):
             if not cleaned_data.get('payment_banktransfer_bank_details'):
                 raise ValidationError(
                     {'payment_banktransfer_bank_details': _('Please enter your bank account details.')})
+
+        if (cleaned_data.get('payment_banktransfer_code_prefix') and not self.code_prefix_patterns.match(cleaned_data.get('payment_banktransfer_code_prefix'))):
+            raise ValidationError(
+                {'payment_banktransfer_code_prefix': _('Prefix can only contains alphanumeric characters and hyphens.')})
```

Comment:
Please use Django's built-in validators instead of doing this yourself:

https://docs.djangoproject.com/en/3.0/ref/validators/#built-in-validators

Usage example:
https://github.com/pretix/pretix/blob/27b56b5aeaa972180f27e4c797dc3042666e3706/src/pretix/control/forms/organizer.py#L225